platform/osgi-platform/src/main/java/org/motechproject/server/osgi/status/PlatformStatusManagerImpl.java

```diff
@@ -34,14 +55,22 @@ public void bundleChanged(BundleEvent bundleEvent) {
         if (bundleEvent.getType() == BundleEvent.STARTED) {
             LOGGER.debug("Bundle {} started", symbolicName);
             platformStatus.addOSGiStartedBundle(symbolicName);
+            checkBundleOsgiStarted(bundleEvent.getBundle());
         } else if (bundleEvent.getType() == BundleEvent.STOPPED) {
             LOGGER.trace("Bundle {} stopped", symbolicName);
             platformStatus.removeOSGiStartedBundle(symbolicName);
+            osgiBundles.remove(bundleEvent.getBundle());
+            blueprintBundles.remove(bundleEvent.getBundle());
+        } else if (bundleEvent.getType() == BundleEvent.UNRESOLVED) {
```

Comment:
Lets skip processing of unresolved for now, I have debug this, and the issue is that MDS refreshing the bundles to use new domain classes will produce this state, which is not an error indicator. I think for now we can just check the stopped osgi event and error blueprint events